git_theta/updates/base.py

```diff
@@ -9,40 +10,148 @@
     from importlib.metadata import entry_points
 
 import logging
-from typing import Optional
+from typing import Optional, Tuple
+
+import numpy as np
 
 from git_theta import git_utils, utils, params, metadata
 
 
-class Update:
-    """Base class for parameter update plugins."""
+Parameter = np.ndarray
 
-    def read(self, param_metadata):
-        lfs_pointer = param_metadata.lfs_metadata.lfs_pointer
-        serialized_param = git_utils.git_lfs_smudge(lfs_pointer)
-        param = params.get_update_serializer().deserialize(serialized_param)
-        return param
 
-    def get_last_version(self, repo, path, param_keys, param_metadata):
-        last_commit = param_metadata.theta_metadata.last_commit
-        logging.debug(f"Getting data from commit {last_commit}")
-        if last_commit:
-            last_metadata_obj = git_utils.get_file_version(repo, path, last_commit)
-            last_metadata = metadata.Metadata.from_file(last_metadata_obj.data_stream)
-            last_param_metadata = last_metadata.flatten()[param_keys]
-            return last_param_metadata
-        else:
-            raise ValueError("Cannot find previous version for parameters")
+class Update(metaclass=ABCMeta):
+    """Base class for parameter update plugins."""
+
+    def __init__(self, serializer: params.Serializer):
+        self.serializer = serializer
 
     @property
-    def name(self):
+    @abstractmethod
+    def name(self) -> str:
         raise NotImplementedError
```

Comment:
Is there any need to raise exception here since the method is abstract?

Reply:
Yep, added docstring (as a body is required) and removed error. Did similar things elsewhere.